--- exper-base/build.sh	2019-05-16 21:27:07.000000000 -0500
+++ temp_build.sh	2019-05-18 11:47:24.000000000 -0500
@@ -29,9 +29,6 @@
   source $outdir/compile.sh
 }
 
-run_reduce() {
-}
-
 run_discharge() {
 }
 
@@ -49,8 +46,6 @@
   record_system $outdir
   echo "Compiling program"
   build $outdir
-  echo "Running reduce"
-  run_reduce $outdir
   echo "Running discharge"
   run_discharge $outdir
   echo "Recording results"
