--- exper-base/build.sh	2019-05-16 21:27:07.000000000 -0500
+++ temp-build.sh	2019-05-17 07:20:12.000000000 -0500
@@ -29,6 +29,11 @@
   source $outdir/compile.sh
 }
 
+get_ring16() {
+  local outdir=$1
+  sed -n -e '/^783$/,/^$/p' >$outdir/ring16.conf RSST/anc/U_2822.conf
+}
+
 run_reduce() {
 }
 
@@ -50,6 +55,7 @@
   echo "Compiling program"
   build $outdir
   echo "Running reduce"
+  get_ring16 $outdir
   run_reduce $outdir
   echo "Running discharge"
   run_discharge $outdir
