diff --git a/build.sh b/build.sh
index e5709e6..58e99ac 100755
--- a/build.sh
+++ b/build.sh
@@ -32,9 +32,6 @@ EOF
 run_reduce() {
 }
 
-run_discharge() {
-}
-
 record_results() {
   local outdir=$1
   git add $outdir
@@ -51,8 +48,6 @@ main() {
   build $outdir
   echo "Running reduce"
   run_reduce $outdir
-  echo "Running discharge"
-  run_discharge $outdir
   echo "Recording results"
   record_results $outdir
 }
